### PR TITLE
Anchor links on liveblog fix

### PR DESCRIPTION
**Pull request by Calanthe**
This PR fixes anchor links on liveblogs when the viewability changes are turned on.

Before:
screen shot 2015-07-30 at 16 51 44

After:
screen shot 2015-07-30 at 16 54 42

It is related to the https://github.com/guardian/frontend/issues/9963

### CODE REVIEW

**Comment by OliverJAsh** [comment]
We're having to add lots of hacks to bring back basic accessibility features like "page down". We had to do something similar previously in #9867 to fix "jump to anchor", and there are more a11y bugs like it.

I'm concerned that adding workarounds like this to our codebase is adding unnecessary complexity. We shouldn't have broken it in the first place.

Thus, my preferred fix would be to remove the sticky nav. :-1: to the overall situation and this solution, but I guess we have no choice.

**Comment by Jholder112233** [comment]
There's another problem as the block level sharing links include an anchor link which will have the same offset problem: http://www.theguardian.com/business/blog/live/2015/jul/30/uk-job-losses-dominate-bumper-results-day-shell-centrica-live#block-55b9fbf6e4b016b93fad24ed

We also have a similar problem with the 'comments' anchor link:
screen shot 2015-07-30 at 17 14 04

**Comment by OliverJAsh** [comment]
Hey Josh, could you file those issues at
https://github.com/guardian/frontend/labels/sticky-nav? Thanks!

On Thu, 30 Jul 2015 at 17:14 Josh Holder <notifications@github.com> wrote:

> There's another problem as the block level sharing links include an anchor
> link which will have the same offset problem:
> http://www.theguardian.com/business/blog/live/2015/jul/30/uk-job-losses-dominate-bumper-results-day-shell-centrica-live#block-55b9fbf6e4b016b93fad24ed
>
> We also have a similar problem with the 'comments' anchor link:
> [image: screen shot 2015-07-30 at 17 14 04]
> <https://cloud.githubusercontent.com/assets/2236852/8988507/694c922a-36de-11e5-8a58-266470d0490e.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/9986#issuecomment-126386778>.
>

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

**Comment by Jholder112233** [comment]
Will do @OliverJAsh!

**Comment by OliverJAsh** [comment]
I've labelled all these bugs with `a11y` and `sticky-nav` for easy reference: https://github.com/guardian/frontend/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3Asticky-nav+label%3Aa11y+

**Comment by Calanthe** [comment]
Yes, I know that such fixes will add some complexity to the codebase.  
But when (and if) we will decide to go fully live with the viewability changes (which does not mean that I like and want sticky ad) at some point we will remove the viewability switch and also every related code (like here condition statement) from the codebase. I think that this is just a temporary solution.

File: static/src/javascripts/bootstraps/liveblog.js [hunk 1]
```diff
@@ -102,8 +102,9 @@ define([
                     eventId = $el.attr('data-event-id'),
                     title = $('.timeline__title', $el).text(),
                     targetEl = qwery('#' + eventId),
-                    dim = bonzo(targetEl).offset();
-                scroller.scrollTo(dim.top - 12, 500, 'easeOutQuint');
+                    dim = bonzo(targetEl).offset(),
+                    scrollAmount = config.switches.viewability ? dim.top - 52 : dim.top - 12;
```
Comment (piuccio) [inline static/src/javascripts/bootstraps/liveblog.js]: Can we put `52`, `12` and `500` in a variable or maybe in `config`. It's hard to understand magic numbers. Why that exact value?

**Comment by johnduffell** [comment]
is this being worked on again? consider closing and reopen when it's ready

**Comment by OliverJAsh** [comment]
I would like to keep all sticky nav issues open for now, so we can easily discuss the remaining issues.

**Comment by gklopper** [comment]
Why can this not be merged?

**Comment by Calanthe** [comment]
@gklopper At the beginning I didn't merge it because I didn't get approval. Later on there was an idea to remove the viewability switch so I wanted to remove the 'config.switches.viewability ?' condition as well. But after all I don't now when exactly we (the commercial team) is going to remove the viewability switch. So I think that this PR can be merged now.

**Pull request merged**